perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -53,43 +54,52 @@ def __init__(self, vm_spec):
         self.key_name = 'perfkit_key_%d_%s' % (self.instance_number,
                                                FLAGS.run_uri)
         self.client = os_utils.NovaClient()
+        # FIXME(meteorfox): Remove --openstack_public_network and
+        # --openstack_private_network once depreciation time has expired
+        self.network_name = (FLAGS.openstack_network or
+                             FLAGS.openstack_private_network)
+        self.floating_ip_pool_name = (FLAGS.openstack_floating_ip_pool or
+                                      FLAGS.openstack_public_network)
         self.public_network = os_network.OpenStackPublicNetwork(
-            FLAGS.openstack_public_network
-        )
+            FLAGS.openstack_floating_ip_pool)
         self.id = None
         self.pk = None
         self.user_name = FLAGS.openstack_image_username
         self.boot_wait_time = None
         self.image = self.image or self.DEFAULT_IMAGE
+        self.public_net = None
+        self.private_net = None
+        self.floating_ip = None
 
     def _Create(self):
         image = self.client.images.findall(name=self.image)[0]
         flavor = self.client.flavors.findall(name=self.machine_type)[0]
+        self.private_net = self.client.networks.find(label=self.network_name)
+        if self.floating_ip_pool_name:
+            self.public_net = self.client.networks.find(
+                label=self.floating_ip_pool_name)
+
+        if self.private_net is None:
```

Comment:
That's a reasonable case, a user could pass an empty string which it would evaluate to False